logstash-core/lib/logstash/plugin.rb

```diff
@@ -45,6 +45,7 @@ def eql?(other)
   end
 
   def initialize(params=nil)
+    @allow_env = params.delete(LogStash::Config::Mixin::ALLOW_ENV_FLAG) { false }
```

Comment:
I understand why we wrote this this way (we both discussed it!), but I think a comment mentioning something like 'this is a temporary hack' would be good. It'd be good to announce our intent to rewrite this weirdness at a later date here :)